orchestration/dap_orchestration/pipelines.py

```diff
@@ -36,19 +36,23 @@
     }
 )
 
+test_mode = ModeDefinition(
+    name="test",
+    resource_defs={
+        "beam_runner": ResourceDefinition.mock_resource(),
+        "refresh_directory": refresh_directory,
+        "outfiles_writer": ResourceDefinition.mock_resource()
+    }
+)
+
 
 @pipeline(
-    mode_defs=[local_mode, dev_mode, prod_mode]
+    mode_defs=[local_mode, dev_mode, prod_mode, test_mode]
 )
 def refresh_data_all() -> None:
-    hles_extract_records()
-    hles_transform_records()
-
-    cslb_extract_records()
-    cslb_transform_records()
-
-    env_extract_records()
-    env_transform_records()
-
-    # the transform output should be in the same directory for all 3 pipelines
-    write_outfiles()
+    collected_outputs = [
+        hles_transform_records(hles_extract_records()),
+        cslb_transform_records(cslb_extract_records()),
+        env_transform_records(env_extract_records())
+    ]
+    write_outfiles(collected_outputs)
```

Comment:
this is how we are enforcing the dependency on the upstream steps using the **fan_in_results** param, right?